File: practice_java_program/notpad/secondSmallest.java
public class secondSmallest{
	public static void main(String[] args){
		int[] numbers = {3,6,7,1,4,2};
		int smallest = Integer.MAX_VALUE;
		int secondSmallest = Integer.MAX_VALUE;
		for(int num:numbers){					//smallest=1    secondsmallest=3
			if(num<smallest){
				secondSmallest=smallest;
				smallest=num;
			}
			else if(num<secondSmallest && num != smallest){     //secondsmallest=2
				secondSmallest=num;
			}
		}
		System.out.println("smallest = "+smallest);
		System.out.println("second smallest = "+secondSmallest);
	}
}

File: python_oriented_interview/notpad/secondSmallest.py
import sys

numbers = [3, 6, 7, 1, 4, 2]
smallest = sys.maxsize
second_smallest = sys.maxsize

for num in numbers:
    if num < smallest:
        second_smallest = smallest
        smallest = num
    elif num < second_smallest and num != smallest:
        second_smallest = num

print("smallest =", smallest)
print("second smallest =", second_smallest)
